crates/apps_lib/src/client/masp.rs

```diff
@@ -107,6 +109,30 @@ pub async fn syncing<
         }};
     }
 
+    #[cfg(feature = "testing")]
+    {
+        // Load the confirmed context (if present) and update the conversions
+        // for the shielded history. This is needed for integration
+        // tests only as the cli wallet is not supposed to compile the
+        // history of shielded transactions
+        let _ = shielded.load_confirmed().await;
+        for (asset_type, (token, denom, position, epoch, _conv)) in
+            namada_sdk::rpc::query_conversions(&client).await?
```

Comment:
@mateuszjasiuk From now on (at least in this version of the shielded history), a call to `query_conversions` will be required before each call to the shield-sync command to prevent it from crashing when trying to make sense of the asset types

Reply:
does this not affect the cli as well?

Reply:
But isn't the history being saved when using the cli? In which case we might hit panics over unrecognized assets. Becuase this code block is always active during tests, I don't know how the cli will behave in real life.